mypy/semanal.py

```diff
@@ -1288,6 +1291,126 @@ def store_declared_types(self, lvalue: Node, typ: Type) -> None:
+        if len(s.lvalues) != 1 or not isinstance(s.lvalues[0], NameExpr):
+            return None
+        if not isinstance(s.rvalue, CallExpr):
+            return None
+        call = s.rvalue
+        if not isinstance(call.callee, RefExpr):
+            return None
+        callee = call.callee
+        if callee.fullname != 'typing.NewType':
+            return None
+        return call
+
+    def get_newtype_name(self, s: AssignmentStmt) -> Optional[str]:
+        lvalue = cast(NameExpr, s.lvalues[0])
+        name = lvalue.name
+        if not lvalue.is_def:
+            if s.type:
+                self.fail("Cannot declare the type of a NewType declaration", s)
+            else:
+                self.fail("Cannot redefine '%s' as a NewType" % name, s)
+            return None
+        return name
+
+    def check_newtype_args(self, call: CallExpr, name: str, context: Context) -> Optional[Type]:
+        has_failed = False
+        args = call.args
+        if len(args) != 2:
+            self.fail("NewType(...) expects exactly two arguments", context)
+            return None
+
+        # Check first argument
+        if not isinstance(args[0], (StrExpr, BytesExpr, UnicodeExpr)):
+            self.fail("Argument 1 to NewType(...) must be a string literal", context)
+            has_failed = True
+        elif call.arg_kinds[0] != ARG_POS:
+            self.fail("Argument 1 to NewType(...) must be a positional string literal", context)
+            has_failed = True
+        elif cast(StrExpr, call.args[0]).value != name:
+            self.fail("Argument 1 to NewType(...) does not match variable name", context)
+            has_failed = True
+
+        # Check second argument
+        try:
+            value = self.anal_type(expr_to_unanalyzed_type(call.args[1]))
+        except TypeTranslationError:
+            self.fail("Argument 2 to NewType(...) must be a valid type", context)
+            return None
+
+        if call.arg_kinds[1] != ARG_POS:
+            self.fail("Argument 2 to NewType(...) must be a positional type", context)
+            has_failed = True
+        elif isinstance(value, Instance) and value.type.is_newtype:
+            self.fail("Argument 2 to NewType(...) cannot be another NewType", context)
+
+        return None if has_failed else value
+
+    def build_newtype_typeinfo(self, name: str, old_type: Type, base_type: Instance) -> TypeInfo:
+        class_def = ClassDef(name, Block([]))
+        class_def.fullname = self.qualified_name(name)
+
+        symbols = SymbolTable()
+        info = TypeInfo(symbols, class_def)
+        info.mro = [info] + base_type.type.mro
+        info.bases = [base_type]
+        info.is_newtype = True
+
+        # Add __init__ method
+        args = [Argument(Var('cls'), NoneTyp(), None, ARG_POS),
+                self.make_argument('item', old_type)]
+        signature = CallableType(
+            arg_types = [cast(Type, None), old_type],
```

Comment:
PEP 8 insists there be no spaces around these `=` signs. Strange that flake8 didn't flag this?